src/main/kotlin/com/openlattice/graph/partioning/RepartitioningJob.kt

```diff
@@ -0,0 +1,352 @@
+INSERT INTO ${DATA.name} SELECT $REPARTITION_DATA_COLUMNS
+    FROM ${DATA.name} INNER JOIN (select ? as ${ENTITY_SET_ID.name},? as ${PARTITIONS.name} ) as es 
+    USING ( ${ENTITY_SET_ID.name} )
+    WHERE ${PARTITION.name} = ? AND ${PARTITION.name}!=$REPARTITION_SELECTOR
+    ON CONFLICT DO UPDATE SET
+        ${latestSql(ORIGIN_ID, VERSION)},
+        ${latestSql(VERSION, VERSION)},
+        $VERSIONS = ARRAY( SELECT DISTINCT UNNEST(${VERSIONS.name} || EXCLUDED.${VERSIONS.name} ) ORDER BY 1  )  
+        ${latestSql(LAST_WRITE, VERSION)},
+        ${latestSql(LAST_PROPAGATE, VERSION)},
+""".trimIndent()
+
+/**
+ * Query for repartition a partition of ids.
+ *
+ * 1. entity set id
+ * 2. partition
+ *
+ * NOTE: Using last_link for LINKING_ID in this query because a link can happen without triggering a version update.
+ */
+private val REPARTITION_IDS_SQL = """
+INSERT INTO ${IDS.name} SELECT $REPARTITION_IDS_COLUMNS
+    FROM ${IDS.name} INNER JOIN (select ? as ${ENTITY_SET_ID.name},? as ${PARTITIONS.name} ) as es 
+    USING (${ENTITY_SET_ID.name})
+    WHERE ${PARTITION.name} = ? AND ${PARTITION.name}!=$REPARTITION_SELECTOR
+    ON CONFLICT DO UPDATE SET
+        ${latestSql(LINKING_ID, LAST_LINK)}, 
+        ${latestSql(VERSION, VERSION)},
+        $VERSIONS = ARRAY( SELECT DISTINCT UNNEST(${VERSIONS.name} || EXCLUDED.${VERSIONS.name} ) ORDER BY 1  ),  
+        ${latestSql(LAST_WRITE, VERSION)},
+        ${latestSql(LAST_INDEX, VERSION)},
+        ${latestSql(LAST_PROPAGATE, VERSION)},
+        ${latestSql(LAST_MIGRATE, VERSION)},
+        ${latestSql(LAST_LINK_INDEX, VERSION)}
+""".trimIndent()
+
+/**
+ * Query for repartition a partition of edges.
+ *
+ * 1. entity set id
+ * 2. partition
+ * NOTE: Using last_link for LINKING_ID in this query because a link can happen without triggering a version update.
+ */
+private val REPARTITION_EDGES_SQL = """
+INSERT INTO ${E.name} SELECT $REPARTITION_EDGES_COLUMNS
+    FROM ${E.name} INNER JOIN (select ? as ${SRC_ENTITY_SET_ID.name},? as ${PARTITIONS.name} ) as es
+    USING (${SRC_ENTITY_SET_ID.name})
+    WHERE ${PARTITION.name} = ? AND ${PARTITION.name}!=$REPARTITION_SELECTOR
+    ON CONFLICT DO UPDATE SET
+        ${latestSql(VERSION, VERSION)},
+        $VERSIONS = ARRAY( SELECT DISTINCT UNNEST(${VERSIONS.name} || EXCLUDED.${VERSIONS.name} ) ORDER BY 1  )
+""".trimIndent()
+
+/**
+ * Computes the actual partition and compares it to current partition. If partitions do not match deletes the row.
+ * 1. entity set id
+ * 2. partition
+ */
+private val DELETE_DATA_SQL = """
+DELETE FROM ${DATA.name} 
+    USING (SELECT ${ID.name},${ENTITY_SET_ID.name},${PARTITION.name} FROM ${DATA.name} INNER JOIN (select ? as ${ENTITY_SET_ID.name},? as ${PARTITIONS.name} ) as es USING (${ENTITY_SET_ID.name})) as to_be_deleted
+    WHERE ${PARTITION.name} = ? AND partition!=$REPARTITION_SELECTOR AND to_be_deleted.${ID.name} = ids.${ID.name} and to_be_deleted.${PARTITION.name} = ids.${PARTITION.name};  
```

Comment:
is this supposed to be `ids` and not `data` ?

Reply:
(and for the edges query)

Reply:
Yup.